### PR TITLE
NAS-3548 add SupportService with availability methods

### PR DESCRIPTION
**Pull request by pcerny**
Two availability methods added. One for check GD.CN is up, another for
wait till GD.CN is up.

### CODE REVIEW

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

File: gooddata-sdk/gooddata_sdk/support.py [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+        """
+        Wait till GD.CN service is available. When timeout is:
+          - > 0 exception is raised after given number of seconds.
+          - = 0 exception is raised whe service is not available immediately
+          - < 0 no timeout
+
+        Method propagates is_available exceptions.
+        :param timeout: seconds to wait to service to be available (see method description for details)
+        """
+        sleep_time_sec = 2.3
```
Comment (jaceksan) [inline gooddata-sdk/gooddata_sdk/support.py:47]: This is magic and not necessary, change to e.g. 2 and transform it into an argument of this method with default 2.

File: gooddata-sdk/gooddata_sdk/support.py [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+        start_time_sec = time.time()
+        while True:
+            if self.is_available:
+                return
+
+            if timeout < 0:
+                time.sleep(sleep_time_sec)
+            else:
+                to_timeout_sec = timeout - (time.time() - start_time_sec)
+                if to_timeout_sec < 0.0001:
```
Comment (jaceksan) [inline gooddata-sdk/gooddata_sdk/support.py:57]: Please, explain here in a comment, why this is necessary (0.0001)

**Review by jaceksan — CHANGES_REQUESTED** [review]

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

File: gooddata-sdk/gooddata_sdk/support.py (hunk 1 above)
Comment (pcerny) [reply gooddata-sdk/gooddata_sdk/support.py]: Done

File: gooddata-sdk/gooddata_sdk/support.py (hunk 2 above)
Comment (pcerny) [reply gooddata-sdk/gooddata_sdk/support.py:60]: Done

**Review by jaceksan — APPROVED** [review]

**Comment by yenkins** [comment]
### Sonar scan result


More detail, see in https://sonarqube-gate.intgdc.com/dashboard?id=gooddata-python-sdk-gate-PR87

To scan for vulnerabilities in dependencies and run unit tests (to get coverage report in sonar) please comment your PR with 'extended check sonar'.

**Comment by gdgate** [comment]
Build succeeded (gate pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-python-sdk-sonarqube-scan-zuul-gate/81/ : SUCCESS in 1m 06s
- https://checklist.intgdc.com/job/compliance/job/gooddata-python-sdk-fossa-scanning-zuul-el7/81/ : SUCCESS in 2m 37s (non-voting)
- https://checklist.intgdc.com/job/compliance/job/gooddata-python-sdk-merge-compliance-check/81/ : SUCCESS in 38s

**Pull request merged**